src/commands/report.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{
+    http,
+    reporter::{self, Report},
+    settings,
+};
+
+use anyhow::Result;
+use cloudflare::framework::{
+    apiclient::ApiClient,
+    endpoint::{Endpoint, Method},
+    response::ApiFailure,
+};
+
+struct ErrorReport(Report);
+
+impl Endpoint<(), (), Report> for ErrorReport {
+    fn method(&self) -> Method {
+        Method::Post
+    }
+
+    fn path(&self) -> String {
+        "wrangler/errors".into()
+    }
+
+    fn body(&self) -> Option<Report> {
+        Some(self.0.clone())
+    }
+}
+
+pub fn run(log: Option<&str>) -> Result<()> {
+    let user = settings::global_user::GlobalUser::new()?;
+    let report = reporter::read_log(log)?;
+    let client = http::cf_v4_client(&user)?;
+    client
+        .request(&ErrorReport(report))
+        .map(|_| ())
+        .map_err(|e| match e {
```

Comment:
This could be better written as 
```rust
if let Some(e) = client
  .request(&ErrorReport(report))
  .err() {
    match e {
      ...
    }
} else {
    Ok(())
}
```
if you wanna discard the success result. Or some variation of that -- `Result::err()` is designed for that case.

Reply:
yea, i tend to prefer mapping when I have to Err() wrap the various internal error values, but i suppose it is a bit more elegant when the Ok is discarded. updated!

Reply:
Thanks! And you're right, these are both totally fine, just a minor nit :)